### PR TITLE
Added commandlet for updating protection job run

### PR DESCRIPTION
**Pull request by bijayprasadtiwari**

### CODE REVIEW

File: src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1 [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#### USAGE ####
+#	Update-CohesityProtectionJobRun -ProtectionJobName viewJob -JobRunId 65675,65163 -ExtendRetention 10
+#	Update-CohesityProtectionJobRun -ProtectionJobName viewJob -JobRunId 65675 -ExtendRetention 10
+#	Update-CohesityProtectionJobRun -ProtectionJobName viewJob -ExtendRetention 10
```
Comment (ashish-cohesity) [inline src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1]: worth mentioning this is in days , also negative reduces the retention by that many days and setting into 0 deletes the snapshot.

Comment (ashish-cohesity) [inline src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1]: Also add starttimeusecs and endtimeusecs as parameter..just like in UI , we can filter job runs between a date range.

File: src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1 [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+					return
+				}
+
+				foreach ($job in $jobResp) {
+					if ($job.Name -eq $ProtectionJobName) {
+						$JobId = $job.id
+						break
+					}
+				}
+			} catch {
```
Comment (ashish-cohesity) [inline src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1]: Good Exception handling!

File: src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1 [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+						$payloadJson = $payload | ConvertTo-Json -Depth 4
+
+						$url = $server + '/irisservices/api/v1/public/protectionRuns'
+						$updateResp = Invoke-RestApi -Method 'Put' -Uri $url -Headers $headers -Body $payloadJson
+						$jobUpdated += 1
+					}
+				}
+			}
+
+			if ($jobUpdated -eq $JobRunId.length) {
```
Comment (ashish-cohesity) [inline src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1]: Would be great to display in format-table the jobruns and extention ids ?

File: src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1 (hunk 1 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1]: Fixed, added comments in the source header

Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1]: Fixed, filter based on the start and end time to update the retention

File: src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1 (hunk 3 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/ProtectionJobRun/Update-CohesityProtectionJobRun.ps1]: Using the getter commandlet to filter and display the final result

**Pull request merged**